Update github.com/spf13/pflag and github.com/spf13/cobra

andreykurilin opened the pull request:
Update github.com/spf13/pflag and github.com/spf13/cobra

Update:
    github.com/spf13/cobra to f62e98d28ab7ad31d707ba837a966378465c7b57
    github.com/spf13/cobra/doc to f62e98d28ab7ad31d707ba837a966378465c7b57
    github.com/spf13/pflag to 1560c1005499d61b80f865c04d39ca7505bf7f0b

Closes issue #29852

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/27855)

<!-- Reviewable:end -->

eparis commented:
ok to test

eparis commented:
ok to test

andreykurilin commented:
@eparis Should I close this pr, since issue was marked as duplicate and it is closed now or it can be merged just like for updating code?

elyscape commented:
IMO, it's still a good idea to update cobra, because that would pull in spf13/cobra#288, which fixes the bash completion for `kubectl create` as described in spf13/cobra#287. In short, it will make `kubectl create <TAB>` suggest `configmap`, `namespace`, `secret`, and `serviceaccount` in additon to `-f` and `--filename=`.

andreykurilin committed:
Update github.com/spf13/pflag and github.com/spf13/cobra

Update:
  github.com/spf13/cobra to f62e98d28ab7ad31d707ba837a966378465c7b57
  github.com/spf13/cobra/doc to f62e98d28ab7ad31d707ba837a966378465c7b57
  github.com/spf13/pflag to 1560c1005499d61b80f865c04d39ca7505bf7f0b

Closes issue #29852

eparis commented:
@andreykurilin looks like there was a slight change in the cobra error message which the _test.go file complains about. Likely we need to update an _test.go as well?

andreykurilin commented:
@eparis yeah, I saw it, but had no time to fix it. I'll try to do it at the beginning of next week

andreykurilin commented:
@eparis, tests are green now

AdoHe commented on cmd/hyperkube/hyperkube_test.go:
```diff
@@ -177,15 +177,15 @@ func TestServerHelp(t *testing.T) {
 	assert.Contains(t, x.output, "A simple server named test1")
-	assert.Contains(t, x.output, "--help[=false]: help for hyperkube")
+	assert.Contains(t, x.output, "-h, --help                                               help for hyperkube")
 	assert.NotContains(t, x.output, "test1 Run")
```
This looks like a little bit strange, the help output really contains so many extra spaces?

andreykurilin replied:
Yes. Here you can find example(several options) of old help:
```
      --global-housekeeping-interval=1m0s: Interval between global housekeepings
      --google-json-key="": The Google Cloud Platform Service Account JSON Key to use for authentication.
      --healthz-bind-address=127.0.0.1: The IP address for the health check server to serve on, defaulting to 127.0.0.1 (set to 0.0.0.0 for all interfaces)
      --healthz-port=10249: The port to bind the health check server. Use 0 to disable.
  -h, --help[=false]: help for hyperkube
      --hostname-override="": If non-empty, will use this string as identification instead of the actual hostname.
      --housekeeping-interval=10s: Interval between container housekeepings
      --iptables-masquerade-bit=14: If using the pure iptables proxy, the bit of the fwmark space to mark packets requiring SNAT with.  Must be within the range [0, 31].
      --iptables-sync-period=30s: How often iptables rules are refreshed (e.g. '5s', '1m', '2h22m').  Must be greater than 0.
```

after this patch it will look like:
[      --global-housekeeping-interval value                 Interval between global housekeepings (default 1m0s)
      --google-json-key string                             The Google Cloud Platform Service Account JSON Key to use for authentication.
      --healthz-bind-address value                         The IP address for the health check server to serve on, defaulting to 127.0.0.1 (set to 0.0.0.0 for all interfaces) (default 127.0.0.1)
      --healthz-port value                                 The port to bind the health check server. Use 0 to disable. (default 10249)
  -h, --help                                               help for hyperkube
      --hostname-override string                           If non-empty, will use this string as identification instead of the actual hostname.
      --housekeeping-interval value                        Interval between container housekeepings (default 10s)
      --iptables-masquerade-bit value                      If using the pure iptables proxy, the bit of the fwmark space to mark packets requiring SNAT with.  Must be within the range [0, 31]. (default 14)
      --iptables-sync-period duration                      How often iptables rules are refreshed (e.g. '5s', '1m', '2h22m').  Must be greater than 0. (default 30s)
](url)

AdoHe replied:
Nice, this looks more user friendly. Thanks @andreykurilin

AdoHe commented:
LGTM.
@eparis kindly ping, please help add the label.

eparis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.kubernetes.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 15 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/27855#-:-KOuPALAbI-UxOimEB1N:bnfp4nl)*
<!-- Sent from Reviewable.io -->

eparis commented:
Reviewed 15 of 15 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/27855)*
<!-- Sent from Reviewable.io -->

andreykurilin commented:
@eparis @AdoHe thanks!
